src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -306,31 +281,17 @@ private BoundIndexerAccess TransformIndexerAccess(BoundIndexerAccess indexerAcce
             ImmutableArray<int> argsToParamsOpt = indexerAccess.ArgsToParamsOpt;
             ImmutableArray<RefKind> argumentRefKinds = indexerAccess.ArgumentRefKindsOpt;
 
-            ImmutableArray<BoundExpression> rewrittenArguments = VisitArguments(
+            ImmutableArray<BoundExpression> rewrittenArguments = VisitArgumentsAndCaptureReceiverIfNeeded(
+                ref transformedReceiver,
+                captureReceiverForMultipleInvocations: CanChangeValueBetweenReads(receiverOpt),
                 indexerAccess.Arguments,
                 indexer,
                 argsToParamsOpt,
                 argumentRefKinds,
-                ref transformedReceiver!,
-                out ArrayBuilder<LocalSymbol>? argumentTemps);
+                stores,
+                ref temps!);
```

Comment:
>! [](http://example.com/codeflow?start=25&length=1)

Minor: Is `!` needed?

Reply:
> Minor: Is ! needed?

Otherwise it complains that method can assign null on the way out